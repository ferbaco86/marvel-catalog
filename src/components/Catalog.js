import React from 'react';
import styled from 'styled-components';
import HeroCard from './HeroCard';
import NavBar from './NavBar';

const CardsContainer = styled.div`
display: flex;
flex-wrap: wrap;
padding: 2rem 6.5rem;
`;

const Catalog = () => (
  <>
    <NavBar />
    <CardsContainer>
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
      <HeroCard />
    </CardsContainer>
  </>
);

export default Catalog;
