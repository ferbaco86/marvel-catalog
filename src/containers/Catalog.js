import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';
import HeroCard from '../components/HeroCard';
import NavBar from '../components/NavBar';
import fetchData from '../api/fetchData';
import LoaderSpinner from '../components/LoaderSpinner';
import Button from '../components/Button';
import ErrorMessage from '../components/ErrorMessage';
import { incrementOffset } from '../actions/index';

const CardsContainer = styled.div`
display: flex;
flex-wrap: wrap;
padding: 2rem 6.5rem;
`;

const Catalog = () => {
  const { data, offset } = useSelector(state => state);
  const dispatch = useDispatch();

  const increaseOffset = () => {
    if (offset.offset < 1500) dispatch(incrementOffset(100));
  };

  useEffect(() => {
    dispatch(fetchData());
    increaseOffset();
  }, [offset]);

  const shouldComponentRender = () => {
    let isPending = false;
    if (data.pending === false || data.error !== null) {
      isPending = false;
    } else {
      isPending = true;
    }
    return isPending;
  };

  const charInfo = data.data;

  if (shouldComponentRender()) return <LoaderSpinner />;
  const errorText = `API Error: ${data.error}`;
  return (
    <>
      <NavBar />
      <CardsContainer>
        {data.error && (
        <ErrorMessage message={errorText} />
        )}
        {charInfo.map(char => (
          <HeroCard
            key={char.id}
            name={char.name}
            image={char.thumbnail.path}
            extension={char.thumbnail.extension}
          />
        ))}
      </CardsContainer>
      {offset.offset < 1500 && (
      <Button name="Next" />)}
    </>
  );
};

export default Catalog;
