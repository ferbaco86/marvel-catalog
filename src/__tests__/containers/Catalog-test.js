import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/index';
import Catalog from '../../containers/Catalog';

describe('Rendering component', () => {
  it('creates a Catalog component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <Catalog />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('LOADING');
  });
});

describe('Display', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Catalog />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
