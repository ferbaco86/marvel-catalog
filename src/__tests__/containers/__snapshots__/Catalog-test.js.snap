// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div
  className="sc-iBPRYJ iPsrfb"
>
  LOADING
  <div
    className="sc-fubCfw byBabO"
  />
</div>
`;
